src/System.Memory/ref/System.Memory.cs

```diff
@@ -87,6 +87,7 @@ public static class SpanExtensions
     {
          public static int IndexOf<T>(this Span<T> span, T value) where T:struct, IEquatable<T> { throw null; }
```

Comment:
That's fine, will slicing + indexof ever be as fast as passing an index and count? I have some doubts about that.

```C#
while (span.Length > 0)
{
      int index = span.IndexOf((byte)'\n');
      if (index == -1) break;
      var line = span.Slice(0, index);
      span = span.Slice(index + 1);
}
```

VS

```C#
int index = 0;
while (index < span.Length)
{
      int lineIndex = span.IndexOf((byte)'\n', index, span.Length - index);
      if (lineIndex == -1) break;
      var line = span.Slice(index, lineIndex - index + 1);
      index += length.Length
}
```

Will that ever be within the ballpark?